### ISSUE TITLE
React 18 support

### ISSUE THREAD

**Issue by adrianmanduc** [zillow/react-slider#257]
Hello. It would be nice to add React v18 to peerDependencies to make using this package easier in projects using React 18. Thanks

**Comment by kris-ellery** [comment]
Hi @adrianmanduc, we need to first test it with React 18 and make sure there are no regressions. We'll update `peerDependencies` accordingly.

**Comment by gabrielmicko** [comment]
Hey @kris-ellery any update on this?

**Comment by DavidChouinard** [comment]
@kris-ellery we're also held up on this to migrate our app to React 18 🙏

**Comment by kris-ellery** [comment]
Added React 18 as a valid `peerDependency`. I tested every documentation example in a small Vite-powered app. However, please report any React 18 incompatibilities.

**Comment by DavidChouinard** [comment]
Fantastic thank you!!

**Comment by wittenator** [comment]
@kris-ellery Could you share the setup of your test app? I just tried adding react-sliders to a fresh Vite react-templated project with React 18 and the builder seems to blow up because of Babble with the error "Could not resolve "@babel/runtime/helpers/extends". If it fits better, I can also open a new issue for this.
